src/towncrier/test/test_build.py

```diff
@@ -407,6 +407,70 @@ def test_no_confirmation(self):
             self.assertFalse(os.path.isfile(fragment_path1))
             self.assertFalse(os.path.isfile(fragment_path2))
 
+    def test_keep_fragments(self):
+        """
+        The `--keep` option will build the full final news file
+        without deleting the fragment files and without
+        any extra CLI interaction or confirmation.
+        """
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
```

Comment:
Could you use the with_isolated_runner decorator in new code please?

https://github.com/twisted/towncrier/blob/trunk/src/towncrier/test/helpers.py#L23

Reply:
done :)